File: Samples/JotFormTest/Classes/Controller/GetAppKeyViewController.m
//
//  GetAppKeyViewController.m
//  JotFormTest
//
//  Created by Interlogy, LLC on 1/5/14.
//  Copyright (c) 2014 Interlogy, LLC. All rights reserved.
//

#import "GetAppKeyViewController.h"
#import "SampleListViewController.h"
#import "SVProgressHUD.h"
#import <JotForm_iOS/JotForm_iOS-Swift.h>

@interface GetAppKeyViewController () {
    JotForm *apiClient;
}

@property (nonatomic,weak) IBOutlet UITextField *usernameTextField;
@property (nonatomic,weak) IBOutlet UITextField *passwordTextField;

@end

@implementation GetAppKeyViewController

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Do any additional setup after loading the view from its nib.
    apiClient = [[JotForm alloc]initWithApiKey:@"" debugMode:NO euApi:NO];
    self.title = @"Get App Key";
    
    [self showAlertView];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:NO];
    self.usernameTextField.text = @"";
    self.passwordTextField.text = @"";
}

#pragma mark - user definition method

- (void)showSampleListViewController {
    SampleListViewController *sampleListVc = [[SampleListViewController alloc] initWithNibName:@"SampleListViewController" bundle:nil];
    
    [self.navigationController pushViewController:sampleListVc animated:YES];
}

- (void)showAlertView {
   /* UIAlertController *alertView = [UIAlertController
                                    alertControllerWithTitle:@"JotFormAPISample"
                                    message:@"Do you have an API key?"
                                    preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *yesButton = [UIAlertAction actionWithTitle:@"Yes"
                                                       style:UIAlertActionStyleDefault
                                                     handler:^(UIAlertAction *action){
                                                         if ([API_KEY isEqualToString:@""]) {
                                                             
                                                             UIAlertController *alertViewCancel = [UIAlertController
                                                                                                   alertControllerWithTitle:@"JotFormAPISample"
                                                                                                   message:@"Please put your API key in Common.h." preferredStyle:UIAlertControllerStyleAlert];
                                                             
                                                             UIAlertAction *cancelButton = [UIAlertAction
                                                                                            actionWithTitle:@"Ok"
                                                                                            style:UIAlertActionStyleDefault
                                                                                            handler:^(UIAlertAction *action){
                                                                                            }];
                                                             [alertViewCancel addAction:cancelButton];
                                                             [self presentViewController:alertViewCancel animated:YES completion:nil];
                                                         } else {
                                                             [[SharedData sharedData] initAPIClient:API_KEY euApi:EU_API];
                                                             
                                                             [self showSampleListViewController];
                                                         }
                                                         
                                                     }];
    
    UIAlertAction *noButton = [UIAlertAction
                                actionWithTitle:@"No"
                                style:UIAlertActionStyleDefault
                                handler:^(UIAlertAction *action){
                                    
                                }];
    
    [alertView addAction:yesButton];
    [alertView addAction:noButton];
    [self presentViewController:alertView animated:YES completion:nil]; */
}

#pragma mark - IBAction

- (IBAction)getAppKeyButtonClicked:(id)sender {
    // Remove cookies to avoid EU API issue.
    NSHTTPCookieStorage *storage = [NSHTTPCookieStorage sharedHTTPCookieStorage];
    for (NSHTTPCookie *cookie in storage.cookies) {
        [storage deleteCookie:cookie];
    }
    
    [self.usernameTextField resignFirstResponder];
    [self.passwordTextField resignFirstResponder];
    
    [SVProgressHUD showWithStatus:@"Getting app key..."];
    
    NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] init];
    userInfo[@"username"] = self.usernameTextField.text;
    userInfo[@"password"] = self.passwordTextField.text;
    userInfo[@"appName"] = @"JotFormAPISample";
    userInfo[@"access"] = @"full";
    
    [apiClient login:userInfo onSuccess:^(id result) {
        if (result) {
            NSInteger responseCode = [result[@"responseCode"] integerValue];
            
            if (responseCode == 200 || responseCode == 206 ) {
                NSString *appKey = result[@"content"][@"appKey"];
                [self checkEuServer:appKey];
            }
        }
    } onFailure:^(id error) {
        [SVProgressHUD dismiss];
    }];
}

- (void)checkEuServer:(NSString *)appKey {
    [apiClient checkEUserver:appKey onSuccess:^(id result) {
        BOOL isEuServer = [result[@"content"][@"euOnly"]boolValue];
        
      //  [[SharedData sharedData] initAPIClient:appKey euApi:isEuServer];
        
        [self showSampleListViewController];
        [SVProgressHUD dismiss];
    } onFailure:^(NSError *error) {
         [SVProgressHUD dismiss];
    }];
}


@end

File: Samples/JotFormTest/Classes/Controller/GetAppKeyViewController.swift
//
//  GetAppKeyViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 1/31/18.
//  Copyright © 2018 wang. All rights reserved.
//

import UIKit
import Foundation
import JotForm_iOS

class GetAppKeyViewController: UIViewController {
    var apiClient: JotForm?
    
    @IBOutlet weak var usernameTextField: UITextField?
    @IBOutlet weak var passwordTextField: UITextField?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        apiClient = JotForm(apiKey: "", debugMode: false, euApi: false)
      
        title = "Get App Key"
        showAlertView()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(false)
        usernameTextField?.text = ""
        passwordTextField?.text = ""
    }
    
    func showSampleListViewController() {
         let sampleListVc = SampleListViewController(nibName: "SampleListViewController", bundle: nil)
        navigationController?.pushViewController(sampleListVc, animated: true)
    }
    
    func showAlertView() {
        let alertView = UIAlertController(title: "JotFormAPISample", message: "Do you have an API key?", preferredStyle: .alert)
        
        let yesButton = UIAlertAction(title: "Yes", style: .default, handler: {(_ action: UIAlertAction) -> Void in
            if (API_KEY == "") {
                let alertViewCancel = UIAlertController(title: "JotFormAPISample", message: "Please put your API key in Common.h.", preferredStyle: .alert)
                
                let cancelButton = UIAlertAction(title: "Ok", style: .default, handler: {(_ action: UIAlertAction) -> Void in
                })
                
                alertViewCancel.addAction(cancelButton)
                self.present(alertViewCancel, animated: true, completion: nil)
            } else {
                SharedData().initAPIClient(API_KEY, euApi: EU_API)
                self.showSampleListViewController()
            }
        })
        
        let noButton = UIAlertAction(title: "No", style: .default, handler: {(_ action: UIAlertAction) -> Void in
        })
        
        alertView.addAction(yesButton)
        alertView.addAction(noButton)
        self.present(alertView, animated: true, completion: nil)
    }
    
    @IBAction func getAppKeyButtonClicked(_ sender: Any) {
        if let cookies = HTTPCookieStorage.shared.cookies {
            for cookie in cookies {
                HTTPCookieStorage.shared.deleteCookie(cookie)
            }
        }
        
        usernameTextField?.resignFirstResponder()
        passwordTextField?.resignFirstResponder()
        
        SVProgressHUD.show(withStatus: "Getting app key...")
        
        var userInfo = [String: Any]()
        userInfo["username"] = usernameTextField?.text
        userInfo["password"] = passwordTextField?.text
        userInfo["appName"] = "JotFormAPISample"
        userInfo["access"] = "full"

        apiClient?.login(userInfo, onSuccess: {(_ result: AnyObject) -> Void in
            let responseCode: Int = result["responseCode"]  as! Int
            if responseCode == 200 || responseCode == 206 {
                let content = result["content"] as AnyObject
                let appKey = content["appKey"] as! String
                self.checkEuServer(appKey)
            } else {
                 SVProgressHUD.dismiss()
            }
        }, onFailure: {(_ error: Any) -> Void in
            SVProgressHUD.dismiss()
        })
    }
    
    func checkEuServer(_ appKey: String) {
        apiClient?.checkEUserver(appKey, onSuccess: {(_ result: AnyObject) -> Void in
            SVProgressHUD.dismiss()
         
            let content = result["content"] as AnyObject
            
            let isEuServer: Bool
            
            if let string = content["euOnly"] as? String {
                isEuServer = (Int(string) ?? 0) != 0
            } else {
                isEuServer = content["euOnly"] as! Bool
            }
            
            SharedData.sharedData.initAPIClient(appKey, euApi:isEuServer)
            
            self.showSampleListViewController()
        }, onFailure: {(_ error: Error?) -> Void in
            SVProgressHUD.dismiss()
        })
    }
}
